src/pages/api/auth-connect/verify-validation-code.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpStatus } from "@fwl/web";
+import { Handler } from "next-iron-session";
+
+import { checkVerificationCode } from "@lib/queries/checkVerificationCode";
+import type { ExtendedRequest } from "@src/@types/ExtendedRequest";
+import { HttpVerbs } from "@src/@types/HttpVerbs";
+import { config } from "@src/config";
+import { MongoNoDataReturned, TemporaryIdentityExpired } from "@src/errors";
+import { withAPIPageLogger } from "@src/middleware/withAPIPageLogger";
+import { withMongoDB } from "@src/middleware/withMongoDB";
+import withSession from "@src/middleware/withSession";
+import { getTemporaryIdentities } from "@src/queries/getTemporaryIdentities";
+import { fetchJson } from "@src/utils/fetchJson";
+import Sentry, { addRequestScopeToSentry } from "@src/utils/sentry";
+
+const handler: Handler = async (request: ExtendedRequest, response) => {
+  addRequestScopeToSentry(request);
+
+  try {
+    if (request.method === "POST") {
+      const { validationCode, eventId } = request.body;
+
+      const user = await getTemporaryIdentities(eventId, request.mongoDb);
+
+      if (user.length === 1 && user[0].temporaryIdentities) {
+        const temporaryIdentity = user[0].temporaryIdentities.find(
+          (temporaryIdentity) => temporaryIdentity.eventId === eventId,
+        );
+
+        if (temporaryIdentity && temporaryIdentity.expiresAt < Date.now()) {
+          const { data } = await checkVerificationCode(validationCode, eventId);
+
+          if (data && data.checkVerificationCode.status === "VALID") {
+            const { value, type } = temporaryIdentity;
+
+            const body = {
+              userId: user[0].sub,
+              type: type.toUpperCase(),
+              value,
+            };
+
+            const route = "/api/identities";
+            const absoluteURL = new URL(route, config.connectDomain).toString();
+
+            await fetchJson(absoluteURL, HttpVerbs.POST, body);
+
+            response.writeHead(HttpStatus.TEMPORARY_REDIRECT, {
+              Location: "/account/logins",
+            });
+
+            response.end();
+          }
```

Comment:
else?

Reply:
Are you ok with just `throw new WrongVerificationCodeInput();` (i.e. not adding a custom error message).

Reply:
Well, for a start, it should reask for the verification code in a loop, so I would imagine a redirect

Reply:
You are right, I totally forgot about that..

Reply:
I'll need to implement the `flash` logic soon imo. Hamza also needs it for the `AlertBar` triggers.